standardization/src/test/scala/za/co/absa/enceladus/standardization/interpreter/StandardizationInterpreter_FractionalSuite.scala

```diff
@@ -0,0 +1,227 @@
+        ErrorMessage.stdCastErr("doubleField", "Infinity"))),
+      FractionalRow("05-Really big", Option(0), Option(reallyBig), Seq(
+        ErrorMessage.stdCastErr("floatField", reallyBig.toString))),
+      FractionalRow("06-NaN", Option(0), Option(0), Seq(
+        ErrorMessage.stdCastErr("floatField", "NaN"),
+        ErrorMessage.stdCastErr("doubleField", "NaN")))
+    )
+
+    val std = StandardizationInterpreter.standardize(src, desiredSchema, "").cache()
+    showDataFrame(std)
+
+    assertResult(exp)(std.as[FractionalRow].collect().sortBy(_.description).toList)
+  }
+
+  test("With infinity from string") {
+    val seq = Seq(
+      ("01-Euler", "2.71", "2.71"),
+      ("02-Null", null, null),
+      ("03-Long", Long.MaxValue.toString, Long.MinValue.toString),
+      ("04-infinity", "-Infinity", "Infinity"),
+      ("05-Really big", "123456789123456791245678912324789123456789123456789.12",
+        "-12345678912345679124567891232478912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912346789123456789123456789123456789123456791245678912324789123456789123456789123456789123456789123456791245678912324789123456789123456789123456789123456789123456789123456789123456789.1"),
+      ("06-Text", "foo", "bar"),
+      ("07-Exponential notation", "-1.23E4", "+9.8765E-3")
+    )
+    val src = seq.toDF("description","floatField", "doubleField")
+    showDataFrame(src)
+
+    val std = StandardizationInterpreter.standardize(src, desiredSchemaWithInfinity, "").cache()
+    showDataFrame(std)
+
+    val exp = Seq(
+      FractionalRow("01-Euler", Option(2.71F), Option(2.71)),
+      FractionalRow("02-Null", Option(0), None, Seq(
+        ErrorMessage.stdNullErr("floatField"))),
+      FractionalRow("03-Long", Option(9.223372E18F), Option(-9.223372036854776E18)),
+      FractionalRow("04-infinity", Some(Float.NegativeInfinity), Option(Double.PositiveInfinity)),
+      FractionalRow("05-Really big", Option(Float.PositiveInfinity), Option(Double.NegativeInfinity)),
+      FractionalRow("06-Text", Option(0), Option(0), Seq(
+        ErrorMessage.stdCastErr("floatField", "foo"),
+        ErrorMessage.stdCastErr("doubleField", "bar"))),
+      FractionalRow("07-Exponential notation", Option(-12300.0f), Option(0.0098765))
+    )
+
+    assertResult(exp)(std.as[FractionalRow].collect().sortBy(_.description).toList)
+  }
+
+  test("With infinity from double") {
+    val reallyBig = Double.MaxValue
+    val seq = Seq(
+      new InputRowDoublesForFractional("01-Euler", Math.E),
+      InputRowDoublesForFractional("02-Null", None, None),
+      InputRowDoublesForFractional("03-Long", Option(Long.MaxValue.toFloat), Option(Long.MinValue.toDouble)),
+      InputRowDoublesForFractional("04-Infinity", Option(Float.NegativeInfinity), Option(Double.PositiveInfinity)),
+      new InputRowDoublesForFractional("05-Really big", reallyBig),
```

Comment:
I don't find this appealing and/or easily understandable. 
+ I would move class definitions to the beginning.

Reply:
You never have unimportant stuff in your code. And these are definitions needed for the understanding of tests. This is like reading a book backwards.

Reply:
If these case classes are used for creating datasets we prefer to keep them in a separate file so several test suites may use them for different purposes. We consider such case classes an example model.